lib/interactor/initializer/helper.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class Interactor::Initializer::Helper
+  METHOD_NAMES = %w[for with run].freeze
+
+  def self.modify_class(target_class, signature, attributes, class_methods)
+    assignments = attr_assignments(attributes)
+
+    target_class.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+        def initialize(#{signature})
+          #{assignments}
+        end
+
+        #{class_methods}
+    RUBY
+
+    attr_readers(target_class, attributes)
+  end
+
+  def self.attr_readers(target_class, attributes)
+    target_class.class_eval do
+      attributes.each do |attr|
+        attr_reader(attr)
+        protected(attr)
+      end
+    end
+  end
+
+  def self.methods_with_params
+    METHOD_NAMES.map do |name|
+      <<-RUBY
+        def self.#{name}(*args)
+            new(*args).run
+          end
+      RUBY
+    end.join("\n")
+  end
+
+  def self.methods_with_keywords
+    METHOD_NAMES.map do |name|
+      <<-RUBY
+        def self.#{name}(*args)
+          new(**args[0]).run
```

Comment:
This looks weird. Maybe you can use the triple dot operator here (`...`)? https://blog.saeloun.com/2019/12/04/ruby-2-7-adds-new-operator-for-arguments-forwarding.html

Reply:
Ah, forgot to change this part after experimentation, but I don't think there's a way to apply the ... operator here.